### PR TITLE
Prevent SB 8-bit stereo DMA from adding zero frames

### PR DESCRIPTION
**Pull request by kcgen**
Thanks to @GranMinigun for flagging case where a stereo 8-bit DMA transfer could add zero frames to the mixer:

``` text
#5  0x000055555591f0f7 in MixerChannel::AddSamples<unsigned char, true, false, true>(unsigned short, unsigned char const*)
    (this=0x55555f947210, frames=0, data=0x55555d0d9848 <sb+40> '\200' <repeats 24 times>) at ../src/hardware/mixer.cpp:776
#6  0x0000555555918dfa in MixerChannel::AddSamples_s8(unsigned short, unsigned char const*)
    (this=0x55555f947210, len=0, data=0x55555d0d9848 <sb+40> '\200' <repeats 24 times>) at ../src/hardware/mixer.cpp:901
#7  0x000055555593f22c in PlayDMATransfer(uint32_t) (bytes_requested=1) at ../src/hardware/sblaster.cpp:746
```

### CODE REVIEW

File: src/hardware/sblaster.cpp [hunk 1]
```diff
@@ -738,15 +738,25 @@ static void PlayDMATransfer(uint32_t bytes_requested)
...
-				sb.chan->AddSamples_s8s(frames,
-				         maybe_silence(samples,
-				                       reinterpret_cast<int8_t *>(sb.dma.buf.b8)));
-			} else {
-				sb.chan->AddSamples_s8(frames,
-				         maybe_silence(samples, sb.dma.buf.b8));
+
+			// Only add whole frames when in a stereo DMA mode. The
+			// number of frames comes from the DMA request and
+			// threfore is data from user-space.
```
Comment (johnnovak) [inline src/hardware/sblaster.cpp:744]: typo: therefore

File: src/hardware/sblaster.cpp [hunk 2]
```diff
@@ -738,15 +738,25 @@ static void PlayDMATransfer(uint32_t bytes_requested)
...
+			LOG_MSG("DMA8 stereo: %d frames", frames);
```
Comment (johnnovak) [inline src/hardware/sblaster.cpp:745]: Should this be debug level?

**Review by johnnovak — APPROVED** [review]
Good catch 👍🏻

File: src/hardware/sblaster.cpp (hunk 2 above)
Comment (kcgen) [reply src/hardware/sblaster.cpp:745]: Accidentally left that in; knocking out. Thanks!

**Comment by kcgen** [comment]
Thanks for the quick review, @johnnovak .

**Pull request merged**